mixxx: install udev rules for HID controllers

ejpcmac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Some udev rules are needed to use Mixxx with a HID controller. This PR makes the rules available so one can add `services.udev.packages = [ pkgs.mixxx ];` to its `configuration.nix` and get the controllers working without any more tweak.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ejpcmac committed:
mixxx: install udev rules for HID controllers

jtojnar commented:
Ideally, upstream would install the udev file for every distro:

https://github.com/mixxxdj/mixxx/blob/bff45d7d3a5ecb4e0856af521e70b3406c4e424f/SConscript#L399-L400

and to the correct path:

https://github.com/mixxxdj/mixxx/blob/bff45d7d3a5ecb4e0856af521e70b3406c4e424f/SConscript#L384

It should be in `libdir` (a.k.a. `unix_lib_path`), `/etc` is reserved for administrators, see https://www.freedesktop.org/software/systemd/man/udev.html#Rules%20Files

ejpcmac commented:
Mmm. I’m not sure to understand completely the SConscript (still in my wake up process): should it work on NixOS? Is it possible for it to work on *every* distro? As for the `postInstall` thing and the path, I’ve taken inspiration from the `openocd` derivation:

https://github.com/NixOS/nixpkgs/blob/84cf00f98031e93f389f1eb93c4a7374a33cc0a9/pkgs/development/tools/misc/openocd/default.nix#L46-L54

Do you want I add a comment to state it is a workaround until upstream has been fixed?

jtojnar commented:
It would need to be modified. I suggest opening an issue (or better pull request) upstream and adding a comment linking to that here.

ejpcmac commented:
I’ve added a comment, changed the location and added an `optionalString` to patch this only for Linux. However, what has to be done upstream is not really clear for me—I’ve never written a SConscript—, so I’m not able to do a quick PR nor write a useful issue. If you have some time, can you please do it as it seems to be clear for you?

jtojnar commented:
Looks like they added CMake script, which installs that by default: https://github.com/mixxxdj/mixxx/blob/e0bca241793926e4d57ad9e8c817fd5d7e0e5c54/CMakeLists.txt#L1146-L1151

ejpcmac commented:
Oh, that’s great news! I’ve updated my comment then, to state the workaround should not be needed for Mixxx 2.3.x.

prusnak commented:
Mixxx 2.3.0 was just released - PR https://github.com/NixOS/nixpkgs/pull/128696

@ejpcmac can you please test whether this PR installs the udev rules correctly?

ejpcmac commented:
@prusnak Sure! I’m a bit busy at the moment so I can’t promise to do this before the week-end, but I’ll try for sure, and update the PR accordingly. Thank you for the update :)

prusnak commented:
The issue should be fixed via https://github.com/NixOS/nixpkgs/pull/128696 which has been just merged. Please open an issue if that is not the case and mixxx is still missing udev rules. Thanks!
